sdk/core/Azure.Core/src/RequestUriBuilder.cs

```diff
@@ -116,7 +117,11 @@ public string Path
             }
         }
 
-        private bool HasQuery => _queryIndex != -1;
+        /// <summary> Gets whether or not this instance of <see cref="RequestUriBuilder"/> has path. </summary>
+        public bool HasPath => _pathAndQuery.Length > 0;
```

Comment:
Curious: Why do `HasPath` and `HasQuery` need to be public?

Reply:
They can be protected.

Reply:
Replace with protected `PathLength` and `QueryLength`